test/new-e2e/tests/installer/host/systemd.go

```diff
@@ -33,39 +35,94 @@ func (h *Host) LastJournaldTimestamp() JournaldTimestamp {
 	return JournaldTimestamp(log.MonotonicTimestamp)
 }
 
+// AssertUnitProperty asserts that the given systemd unit has the given property
+func (h *Host) AssertUnitProperty(unit, property, value string) {
+	res, err := h.remote.Execute(fmt.Sprintf("sudo systemctl show -p %s %s", property, unit))
+	require.NoError(h.t, err)
+	require.Equal(h.t, fmt.Sprintf("%s=%s\n", property, value), res)
+}
+
+func popIfMatches(searchedEvents []SystemdEvent, log journaldLog) []SystemdEvent {
+	for i, event := range searchedEvents {
+		if eventMatches(log, event) {
+			newEvents := make([]SystemdEvent, 0, len(searchedEvents)-1)
+			newEvents = append(newEvents, searchedEvents[:i]...)
+			newEvents = append(newEvents, searchedEvents[i+1:]...)
+			return newEvents
+		}
+	}
+	return searchedEvents
+}
+
+func eventMatches(log journaldLog, event SystemdEvent) bool {
+	if log.Unit != event.Unit {
+		return false
+	}
+	match, _ := regexp.MatchString(event.Pattern, log.Message)
+	return match
+}
+
+func (h *Host) stripSystemd244(events []SystemdEvent) []SystemdEvent {
+	if h.systemdVersion >= 244 {
+		return events
+	}
+	newEvents := make([]SystemdEvent, 0, len(events))
+	map244Units := map[string]struct{}{
+		"datadog-agent-sysprobe.service":     {},
+		"datadog-agent-security.service":     {},
+		"datadog-agent-sysprobe-exp.service": {},
+		"datadog-agent-security-exp.service": {},
+	}
+	for _, e := range events {
+		if _, ok := map244Units[e.Unit]; ok {
+			continue
+		}
+		newEvents = append(newEvents, e)
+	}
+	return newEvents
+}
+
 // AssertSystemdEvents asserts that the systemd events have been logged since the given timestamp
 func (h *Host) AssertSystemdEvents(since JournaldTimestamp, events SystemdEventSequence) {
+	var lastSearchedEvents []SystemdEvent
 	success := assert.Eventually(h.t, func() bool {
 		logs := h.journaldLogsSince(since)
 		if len(logs) < len(events.Events) {
 			return false
 		}
-		i := 0
-		j := 0
+		i, j := 0, 0
+		var searchedEvents []SystemdEvent
 		for i < len(logs) && j < len(events.Events) {
-			if logs[i].Unit == events.Events[j].Unit && strings.HasPrefix(logs[i].Message, events.Events[j].MessagePrefix) {
+			if len(searchedEvents) == 0 {
+				searchedEvents = events.Events[j]
 				j++
 			}
+			searchedEvents = h.stripSystemd244(searchedEvents)
```

Comment:
can you add a comment or name the function more explicitly, without blaming I don't think anyone is going to understand what's going on there